speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -39,162 +40,445 @@
+            self.model = override_hf_model_partial_fn(self.model)
+
+        # Set norm flag for input
+        if norm_input is not None:
+            if type(norm_input) is str:
+                self.norm_input = eval(f"self.model.{norm_input}")
+            elif type(norm_input) is bool:
+                self.norm_input = norm_input
+            else:
+                raise ValueError(f"norm_input should be bool or str")
+        else:
+            self.norm_input = False
+
+        # Set inner forward function
+        if forward_partial_fn is not None:
+            if isinstance(forward_partial_fn, partial):
+                self.forward_partial_fn = forward_partial_fn
+                self.forward_partial_fn.keywords["model"] = self.model
+            else:
+                self.forward_partial_fn = partial(
+                    forward_partial_fn, model=self.model
+                )
+        else:
+            self.forward_partial_fn = partial(default_forward, model=self.model)
+
+        # Set norm flag for output
+        if norm_output is not None:
+            if type(norm_output) is str:
+                self.norm_output = eval(f"self.model.{norm_output}")
+            elif type(norm_input) is bool:
+                self.norm_output = norm_output
+            else:
+                raise ValueError(f"norm_output should be bool or str")
+        else:
+            self.norm_output = False
 
-        # We check if inputs need to be normalized w.r.t pretrained wav2vec2
-        self.normalize_wav = self.feature_extractor.do_normalize
+        self.forward_returns_tuple = forward_returns_tuple
 
+        # Prepare for training, fine-tuning, or inference
         self.freeze = freeze
-        self.freeze_feature_extractor = freeze_feature_extractor
-        self.output_norm = output_norm
         if self.freeze:
             logger.warning(
-                "speechbrain.lobes.models.huggingface_wav2vec - wav2vec 2.0 is frozen."
+                "speechbrain.lobes.models.HuggingFaceModel is frozen."
             )
             self.model.eval()
             for param in self.model.parameters():
                 param.requires_grad = False
         else:
+            self.model.gradient_checkpointing_disable()  # Required by DDP
             self.model.train()
-            if self.freeze_feature_extractor:
-                self.model.feature_extractor._freeze_parameters()
-        self.output_all_hiddens = output_all_hiddens
-
-    def _from_pretrained(self, source, config, model, save_path):
+            if freeze_nested_models_their_calls is not None:
+                if type(freeze_nested_models_their_calls) is list:
+                    freeze_nested_models_their_calls = [
+                        freeze_nested_models_their_calls
+                    ]
+                for nested_freeze_call in freeze_nested_models_their_calls:
+                    eval(f"self.model.{nested_freeze_call}()")
+
+    def _from_pretrained(
```

Comment:
documentation on args

Reply:
actually, it is within a class - we have barely that detail level there on funcs in other classes...

Reply:
Compare https://github.com/speechbrain/speechbrain/blob/261cecf162766e04148cbd9c33eb7d58064f03e0/speechbrain/lobes/models/EnhanceResnet.py#L114

so, no args documentation here as well